homeassistant/components/sensor/ads.py

```diff
@@ -0,0 +1,92 @@
+"""Support for ADS sensors."""
+
+import logging
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_UNIT_OF_MEASUREMENT
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components import ads
+from homeassistant.components.ads import CONF_ADSVAR, CONF_ADSTYPE, \
+    CONF_ADS_FACTOR
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'ADS sensor'
+DEPENDENCIES = ['ads']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADSVAR): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT, default=''): cv.string,
+    vol.Optional(CONF_ADSTYPE, default=ads.ADSTYPE_INT): vol.In(
+        [ads.ADSTYPE_INT, ads.ADSTYPE_UINT, ads.ADSTYPE_BYTE]
+    ),
+    vol.Optional(CONF_ADS_FACTOR): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up an ADS sensor device."""
+    ads_hub = hass.data.get(ads.DATA_ADS)
+    if not ads_hub:
+        return False
+
+    adsvar = config.get(CONF_ADSVAR)
+    adstype = config.get(CONF_ADSTYPE)
+    name = config.get(CONF_NAME)
+    unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
+    factor = config.get(CONF_ADS_FACTOR)
+
+    entity = AdsSensor(ads_hub, adsvar, adstype, name,
+                       unit_of_measurement, factor)
+
+    add_devices([entity])
+
+    ads_hub.add_device_notification(adsvar, ads_hub.ADS_TYPEMAP[adstype],
```

Comment:
Move this to `async_added_to_hass` on the entity.

Reply:
Done.